File: javaFinal/src/javaFinal/IceCreamCone.java
package javaFinal;

public class IceCreamCone extends IceCream {
	
	private String type; //cup or cone
	
	public IceCreamCone(String flavor, String type) {
		super(flavor);
		this.type = type;
	}
	
	@Override
	public void addTopping(String topping) {
		if (topping.equals("banana")) {
			System.out.println("sorry msg - no banana");
		}
		else {
			super.addTopping(topping);
		}
	}
	
	@Override
	public void display() {
		System.out.println("This ice cream is in a " + type);
		super.display();
	}
	
}

File: cppFinal/IceCreamCone.cpp
#include "IceCreamCone.h"
#include <iostream>
using namespace std;

IceCreamCone::IceCreamCone(string flavor_, string type_) 
: type(type_), IceCream(flavor_) {

}

void IceCreamCone::addTopping(string topping) {
    if (topping.compare("banana") == 0) {
        cout << "sorry msg - no banana" << endl;
    }
    else {
        IceCream::addTopping(topping);
    }
}

void IceCreamCone::display() {
    cout << "This ice cream is in a " << type << endl;
    IceCream::display();
}

IceCreamCone::~IceCreamCone() {

}